controller/jiminy/src/account_utils.rs

```diff
@@ -18,10 +18,14 @@ const _ACC_DATA_ALIGN: usize = 8;
 const _POOL_STATE_ALIGN_CHECK: () = assert!(core::mem::align_of::<PoolState>() <= _ACC_DATA_ALIGN);
 
 #[inline]
-pub fn pool_state_checked(acc: &Account) -> Result<&PoolState, Inf1CtlCustomProgErr> {
+pub fn pool_state_of_acc_data(data: &[u8]) -> Option<&PoolState> {
     // safety: account data is 8-byte aligned
-    unsafe { PoolState::of_acc_data(acc.data()) }
-        .ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidPoolStateData))
+    unsafe { PoolState::of_acc_data(data) }
+}
+
+#[inline]
+pub fn pool_state_checked(acc: &Account) -> Result<&PoolState, Inf1CtlCustomProgErr> {
+    pool_state_of_acc_data(acc.data()).ok_or(Inf1CtlCustomProgErr(Inf1CtlErr::InvalidPoolStateData))
 }
```

Comment:
why the split into 2 fns here? We always want it to return `InvalidPoolStateData` when it fails right?